wagtail/admin/views/bulk_action.py

```diff
@@ -53,14 +53,16 @@ def get_queryset(cls, object_ids):
     def check_perm(self, obj):
         return True
 
-    @classmethod
-    def execute_action(cls, objects):
+    def execute_action(self, objects):
         raise NotImplementedError("execute_action needs to be implemented")
 
     @classmethod
     def execute(cls, object_ids):
+        # this method can be called to execute the action without instantiating the class.
```

Comment:
This feels like quite a confusing way of doing this to me: this should either be a classmethod, or it shouldn't be. What you could do is keep it a classmethod, but pass in the data it needs through keyword arguments, for instance.